agents/research_agent.py

```
@@ -8,224 +8,119 @@
                 timeout=self.timeout
             )
             
-            research_content = ""
-            sources = []
+            content = response.choices[0].message.content
+            cost = 0.0
+            
+            # Extract cost and routing information from Router
+            if hasattr(response, '_hidden_params'):
+                cost = response._hidden_params.get('response_cost', 0.0)
+                cost_source = response._hidden_params.get('cost_source', 'openai')
```

Ruff (0.12.2)

Local variable `cost_source` is assigned to but never used

Remove assignment to unused variable `cost_source`

(F841)